presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/ArrayAggregationState.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import io.airlift.slice.SliceOutput;
+
+import com.facebook.presto.spi.type.Type;
+
+@AccumulatorStateMetadata(stateFactoryClass = ArrayAggregationStateFactory.class, stateSerializerClass = ArrayAggregationStateSerializer.class)
+public interface ArrayAggregationState
+    extends AccumulatorState
+{
+    Type getType();
+    SliceOutput getSliceOutput();
+    void setSliceOutput(SliceOutput value);
```

Comment:
Instead of using a SliceOutput, change this to be a `BlockBuilder` and make it work similarly to `MapAggregation` which uses `KeyValuePairs`. This will make it easier to update when we land https://github.com/facebook/presto/pull/2240